htmresearch/frameworks/classification/classification_network.py

```diff
@@ -446,39 +472,68 @@ def trainNetwork(network, networkConfig, networkPartitions, numRecords,
                      avgPredictedInactiveCols,
                      avgUnpredictedActiveCols
                    ))
         _LOGGER.info(tmStats)
 
-        _LOGGER.info(sensorRegion.getOutputData("categoryOut")[0])
-        _plotTMActivation(tm, sensorValueTrace, numRecordsToPlot=100)
+    if recordNumber == partitions[0][1]:
+      # end of the current partition
+      partitionName = partitions[0][0]
+
+      # stop learning for all regions
+      if partitionName == TEST_PARTITION_NAME:
+        _stopLearning(network, trainedRegionNames, recordNumber)
+
+      else:
+        partitions.pop(0)
+        trainedRegionNames.append(partitionName)
+        _enableRegionLearning(network,
+                              trainedRegionNames,
+                              partitionName,
+                              recordNumber)
 
     if recordNumber >= partitions[-1][1]:
       # evaluate the predictions on the test set
       # classifierConfig = networkConfig["classifierRegionConfig"]
       classifierRegion.setParameter("inferenceMode", True)
 
-      actualValue = sensorRegion.getOutputData("categoryOut")[0]
-      inferredValue = _getClassifierInference(classifierRegion)
-      if actualValue == inferredValue:
-        numCorrect += 1
-      else:
-        if verbosity > 0:
-          _LOGGER.debug("recordNum=%s, actualValue=%s, inferredValue=%s"
-                        % (recordNumber, actualValue, inferredValue))
+      if actualCategory == inferredCategory:
+        numCorrectlyClassifiedTestRecords += 1
+
       numTestRecords += 1
-      classificationAccuracy = round(100.0 * numCorrect / numTestRecords, 2)
-      results = (
-      "RESULTS: accuracy=%s | %s correctly classified records out of %s "
-      "test records \n" % (classificationAccuracy,
-                           numCorrect,
-                           numTestRecords))
-  _LOGGER.info(results)
+      testClassificationAccuracy = round(
+        100.0 * numCorrectlyClassifiedTestRecords / numTestRecords, 2)
+      testClassificationAccuracyTrace.append(testClassificationAccuracy)
+
+  _LOGGER.info("RESULTS: accuracy=%s | "
+               "%s correctly classified records out of %s test records \n" %
+               (testClassificationAccuracyTrace[-1],
+                numCorrectlyClassifiedTestRecords,
+                numTestRecords))
 
-  return classificationAccuracy
+  traces = {
+    'classificationAccuracyTrace': classificationAccuracyTrace,
+    'testClassificationAccuracyTrace': testClassificationAccuracyTrace,
+    'sensorValueTrace': sensorValueTrace,
+    'categoryTrace': categoryTrace ,
+    'tmActiveCellsTrace': tmActiveCellsTrace,
+    'tmPredictiveActiveCellsTrace': tmPredictedActiveCellsTrace,
+    'tpActiveCellsTrace': tpActiveCellsTrace
+  }
+
+  if trackTMmetrics:
+    traces['activeColsTrace'] = activeColsTrace.data
+    traces['predictedActiveColsTrace'] = predictedActiveColsTrace.data
+
+    _plotTMActivation(tm, sensorValueTrace,
```

Comment:
I wonder whether it is useful to have this plotting function at all. The plots won't show up on my computer, and the figures are not saved anywhere